content-services/content-sources-backend
# HMS-4747: automatic deletion of outdated snapshots

## Summary
This PR adds automatic deletion of outdated snapshots to the `nightly-jobs` command. The date after which the snapshots should be deleted is settable in the config.
Also adds a list of snapshots which are going to be deleted in the next 14 days to the fetch/list template endpoints.

## Testing steps
1. Create a repository and let it snapshot.
2. Create a template that uses the repo and is using the latest snap.
3. Connect to the DB CLI `make db-cli-connect` and set the snapshot date to the past.
    - `UPDATE snapshots SET created_at = (NOW() - CAST('400 days' AS INTERVAL)) WHERE uuid = '$SNAPSHOT_UUID';`
4. Run nightly jobs (`go run cmd/external-repos/main.go nightly-jobs`) and verify that there wasn't a "delete-snapshots" task enqueued.
5. Verify the outdated snaphot is listed in both the "snapshots" and "to_be_deleted_snapshots" when querying templates.
6. Edit the repo so that it snapshots again and change the template to use the old one.
7. Run the nightly jobs again and verify the snapshot has been deleted and the template updated to use the other one.
